client/src/components/organisms/AdminRequestGroupList.tsx

```diff
@@ -0,0 +1,130 @@
+import { bindActionCreators, Dispatch } from "redux"
+import { gql, useQuery } from "@apollo/client";
+import React, { FunctionComponent, useState } from "react";
+import { connect } from "react-redux";
+
+import { loadRequestGroups, setDisplayRequestGroups } from '../../data/actions/requestGroupsActions';
+import { Button } from "../atoms/Button";
+import RequestGroup from '../../data/types/requestGroup';
+import RequestGroupTable from "../molecules/RequestGroupTable";
+import { RootState } from '../../data/reducers'
+import SearchBar from "../atoms/SearchBar";
+import SimplePageNavigation from "../atoms/SimplePageNavigation";
+
+interface StateProps {
+    requestGroups: Array<RequestGroup>,
+    displayRequestGroups: Array<RequestGroup>
+}
+
+interface DispatchProps {
+    loadRequestGroups: typeof loadRequestGroups,
+    setDisplayRequestGroups: typeof setDisplayRequestGroups
+}
+
+type Props = StateProps & DispatchProps;
+
+const AdminRequestGroupList: FunctionComponent<Props> = (props: React.PropsWithChildren<Props>) => {
+    const [currentPage, setCurrentPage] = useState(1); // Indexing starting at 1.
+    const numGroupsPerPage = 20;
+
+    const handlePageChange = (newPage: number) => {
+        setCurrentPage(newPage);
+    };
+
+    const query = gql`
+    {
+      requestGroups {
+        name
+        dateUpdated
+        deleted
+        description
+        requirements
+        image
+        numOpen
+        nextRecipient {
+          fullName
+        }
+        requestTypes {
+          name
+          deleted
+        }
+        hasAnyRequests
+      }
+    }`
+
+    const sortRequestGroupsAlphabetically = (requestGroups: Array<RequestGroup>) => requestGroups.sort((rg1: RequestGroup, rg2: RequestGroup) => {
+        if (rg1.name && rg2.name && rg1.name < rg2.name) { return -1; }
+        if (rg1.name && rg2.name && rg1.name > rg2.name) { return 1; }
+        return 0;
+    });
+
+    useQuery(query, {
+        onCompleted: (data: { requestGroups: Array<RequestGroup> }) => {
+            // sort fetched request groups alphabetically and filter out deleted request groups
+            const displayRequestGroups = sortRequestGroupsAlphabetically(data.requestGroups.map(requestGroup => ({ ...requestGroup }))).filter(requestGroup => !requestGroup.deleted);
+            props.loadRequestGroups(displayRequestGroups);
+            props.setDisplayRequestGroups(displayRequestGroups);
+        },
+    });
+
+    const onSearchStringChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+        setCurrentPage(1); // when search string changes, reset pagination
+        let updatedRequestGroups = [];
+        if (event.target.value.length > 0) {
```

Comment:
For speed, please only setDisplayRequestGroups when event.target.value.length > 0

Reply:
guy and i talked about this, but just to leave a record here, the reason we are setting setDisplayRequestGroups when the search string is empty is because we want to reset the display request groups to all groups when the user clears their search